### ISSUE TITLE
Problem about different tflite model's invoke time cost?

### ISSUE THREAD

**Issue by LeopoldACC** [tensorflow/tflite-support#247]
I succesfully convert a object detect model and a semantic segmentation keras model(.h5) into tensorflowlite model(.tflite) ,but i found that when i did inference,the segmentation model cost more than 4 seconds to invoke() but object model just need less than 0.1seconds

below is inference code of segmentation model 
```
import numpy as np
import cv2
import tensorflow as tf
import time
import os
def preprocess(x):
    x = x.astype(np.float32)
    x /= 255.0
    mean = [0.485, 0.456, 0.406]
    std = [0.229, 0.224, 0.225]
    x[..., 0] -= mean[0]
    x[..., 1] -= mean[1]
    x[..., 2] -= mean[2]
    if std is not None:
        x[..., 0] /= std[0]
        x[..., 1] /= std[1]
        x[..., 2] /= std[2]
    return x
# Load TFLite model and allocate tensors.
interpreter = tf.lite.Interpreter(model_path="model_segment_vl_512x1024.tflite")
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

# Test model on random input data.
input_shape = input_details[0]['shape']

height = 1024
width = 512
path = './images/5x/VL_5x/'
img_list = os.listdir(path)
st = time.time()
for i,img_name in enumerate(img_list):
    if i>4:
        break
    img_path = path + img_name
    I = preprocess(cv2.imread(img_path))#[:,64:576,:]
    I0 = cv2.resize(I, (width, height))
    
    interpreter.set_tensor(input_details[0]['index'], I0.reshape((1, width, height, 3)).astype(np.float32))
    
    interpreter.invoke()
   
    # The function `get_tensor()` returns a copy of the tensor data.
    # Use `tensor()` in order to get a pointer to the tensor.
    
    output_data = interpreter.get_tensor(output_details[0]['index'])
     
    # print(output_data)

    # cv2.imshow('test0', np.squeeze(output_data))
    # cv2.waitKey(-1)
print(time.time()-st)    

```
5 images cost 22.35933494567871s,avg time cost ≈4.4s

below is inference code of object detect model 
```
import tensorflow as tf

from math import pi
import time
import numpy as np
import cv2

weights_name = 'model_VL.tflite'
w = 224
h = 224

interpreter = tf.lite.Interpreter(model_path=weights_name)
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
image = cv2.imread('images/3x/VL_3x/0020430.png')
image_out = image
H = image_out.shape[0]
DW = int((image_out.shape[1]-image_out.shape[0])/2)
DH = 0
count = 0
st = time.time()
for i in range(3):
    O = cv2.resize(image[DH:DH+H,DW:DW+H,:], (w, h))
    
    interpreter.set_tensor(input_details[0]['index'], 
        O.reshape((1, h, w, 3)).astype(np.float32))
    interpreter.invoke()        
    Y = interpreter.get_tensor(output_details[0]['index'])

    x = int(round((Y[0][2]/2+0.25)*H))+DW
    y = int(round((Y[0][1]/2+0.25)*H))+DH
    H = int(round(H*0.3))
    if Y[0][0]>0.5:
        image_out = cv2.rectangle(image_out, (x-H, y-H), (x+H, y+H), (0, 0, 255), 5)
        count += 1
    DW = x-H
    DH = y-H
    H = H*2
    if DW<0:
        DW = 0
    if DW + H>=image_out.shape[1]:
        DW = image_out.shape[1] - H
    if DH<0:
        DH = 0
    if DH + H>=image_out.shape[0]:
        DH = image_out.shape[0] - H
print(time.time()-st)

```
3 images cost 0.26386356353759766s,avg cost ≈0.09s

How can i speed up my segmentation model' invoke()?
Thanks!

**Comment by lu-wang-g** [comment]
@LeopoldACC You can try the following ways to optimize the performance.
(1) Use a model with smaller input size. The one you are using is 512x1024, which is pretty large. 
(2) Config Interpreter with multithread, see the example [here](https://github.com/tensorflow/examples/blob/408abb536005d697219da5f71f6259e23ae9b512/lite/examples/image_segmentation/android/lib_interpreter/src/main/java/org/tensorflow/lite/examples/imagesegmentation/tflite/ImageSegmentationModelExecutor.kt#L155)
(3) Use GPU delegate. See the instruction [here](https://www.tensorflow.org/lite/performance/gpu)

Hope it helps.

**Comment by LeopoldACC** [comment]
In my task, the segmentation object is thin and thin(It's **one or two pixels** wide at 1920 * 1080 resolution). If I continue down sampling, the information of this object may be lost, and the accuracy will inevitably decline seriously. If I cut 256 * 256 blocks from 1024 * 512, I need to execute invoke () 8 times for an image. This task is really difficult for me to achieve real-time and high precision.But thank you a lot for your timely feedback!

**Comment by lu-wang-g** [comment]
Usually the segmentation model will be applied on a smaller size of image such as 316x316 and then upscale the output mask to the original image size. In that way, you'll get a much faster inference.

**Comment by lu-wang-g** [comment]
Closing this issue. Feel free to reopen it if you have other questions.
